examples/mdetr/matcher.py

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import List, Tuple
+
+import torch
+import torch.nn.functional as F
+from scipy.optimize import linear_sum_assignment
+from torch import nn, Tensor
+from torchvision.ops.boxes import box_convert, generalized_box_iou
+
+
+class HungarianMatcher(nn.Module):
```

Comment:
Why putting this in examples not in `torchmultimodal/modules/loss/`?

Reply:
It is pretty specific to MDETR in that it is performing a bipartite matching between predicted bounding boxes and ground truth bounding boxes. So it is not really a loss per se, even though it is used as part of the final loss calculation. Also, we already have to put one of the MDETR losses in examples (see #138), so we do not lose anything by having this under examples.